.github/workflows/workflow.yml

```diff
@@ -0,0 +1,50 @@
+name: Java CI
+
+on: [push]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    name: maven test and package
+
+    steps:
+    - uses: actions/checkout@v1
+    - name: Set up JDK 1.8
+      uses: actions/setup-java@v1
+      with:
+        java-version: 1.8
+    - name: Build with Maven
+      run: mvn -B package --file pom.xml
+
+  github_release:
+    if: startsWith(github.ref, 'refs/tags/v')
+    name: Create GitHub Release
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+      - name: Set up JDK 1.8
+        uses: actions/setup-java@v1
+        with:
+          java-version: 1.8
+      - name: Build with Maven
+        run: mvn -B package --file pom.xml
+
+      - name: Generate release notes
```

Comment:
not sure if you want this or not, it uses this https://github.com/git-chglog/git-chglog to generate releases with changelogs as part of the release

Reply:
probably not but I should look more into it to see what it offers. Most of the changes for simple clients are relatively small and simplistic because it's a pretty low level tool